test/e2e2/spec/test_api_rr.py

```diff
@@ -166,15 +165,16 @@ def test_soundness():
 
 
 @pytest.mark.parametrize("rr_type,value", generate_params(VALID_RECORDS_CANONICAL))
-def test_create_valid_canonical(api_user_domain: DeSECAPIV1Client, ns_lord: NSClient, rr_type: str, value: str):
+def test_create_valid_canonical(api_user_domain: DeSECAPIV1Client, rr_type: str, value: str):
     assert api_user_domain.rr_set_create(api_user_domain.domains[0], rr_type, [value], subname="a").status_code == 201
-    assert ns_lord.query(f"a.{api_user_domain.domains[0]}", rr_type) == {value}
+    assert NSLordClient.query(f"a.{api_user_domain.domains[0]}", rr_type) == {value}
+    assert_eventually(lambda: query_replication(api_user_domain.domains[0], "a", rr_type) == {value})
 
 
 @pytest.mark.parametrize("rr_type,value", generate_params(VALID_RECORDS_NON_CANONICAL))
-def test_create_valid_non_canonical(api_user_domain: DeSECAPIV1Client, ns_lord: NSClient, rr_type: str, value: str):
+def test_create_valid_non_canonical(api_user_domain: DeSECAPIV1Client, rr_type: str, value: str):
     assert api_user_domain.rr_set_create(api_user_domain.domains[0], rr_type, [value], subname="a").status_code == 201
-    assert len(ns_lord.query(f"a.{api_user_domain.domains[0]}", rr_type)) == 1
+    assert len(NSLordClient.query(f"a.{api_user_domain.domains[0]}", rr_type)) == 1
 
```

Comment:
Shall we not also call `query_replication()` and check that the record is at least there?

Reply:
I rebased much of those tests onto master, does this comment still apply? Where is the new location in the rebased code?

Reply:
Also, unrelated, is here an 'eventually' missing?

Reply:
Regarding your first question: At the end of `test_create_valid_canonical`, [we check that replication worked](https://github.com/desec-io/desec-stack/blob/253837b/test/e2e2/spec/test_api_rr.py#L362), but we don't do that at the end of `test_create_valid_non_canonical`. It looks like you found the code location already.

Regarding the "eventually": I don't think so -- we're querying nslord, which we provision with the RRset in a _synchronous_ pdns API request two lines further up. As soon as that returns, the RRset should be queryable, and not `_eventually` should be needed.

Reply:
fixed by @peterthomassen in 137c2b904cc72092e8bd17ab3e2be7e1f8cfa788